### PR TITLE
Fix baddbmm and scalar_tensor

### PR DESCRIPTION
**Pull request by justinchuby**
1. Handle baddbmm when scalars are SymFloat
2. Accept bool as scalar_tensor input

Fixes https://github.com/justinchuby/torch-onnx/issues/42

### CODE REVIEW

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 1]
```diff
@@ -1017,20 +1017,27 @@ def reshape_to_3d(tensor):
...
-@torch_op("aten::baddbmm")
+@torch_op("aten::baddbmm", trace_only=True)
 def aten_baddbmm(
     self: TRealOrUInt8,
     batch1: TRealUnlessInt16OrInt8,
     batch2: TRealUnlessInt16OrInt8,
-    beta: float = 1.0,
-    alpha: float = 1.0,
+    beta: Optional[TFloat],
+    alpha: Optional[TFloat],
```
Comment (titaiwangms) [inline onnxscript/function_libs/torch_lib/ops/core.py:1026]: default to None?

**Review by titaiwangms — APPROVED** [review]

File: onnxscript/function_libs/torch_lib/ops/core.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py:1026]: Good catch, thanks.

Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:1026]:
```suggestion
    beta: Optional[TFloat] = None,
    alpha: Optional[TFloat] = None,
```

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 2]
```diff
@@ -1017,20 +1017,27 @@ def reshape_to_3d(tensor):
...
+    if alpha is None:
+        # alpha is 1
+        mul_a = batch_mul
+    else:
+        mul_a = op.Mul(batch_mul, op.CastLike(alpha, self))
+    if beta is None:
+        # beta is 1
+        mul_b = self
+    else:
+        mul_b = op.Mul(self, op.CastLike(beta, self))
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:1040]:
```suggestion
    if alpha is None or alpha == 1:
        mul_a = batch_mul
    else:
        mul_a = op.Mul(batch_mul, op.CastLike(alpha, self))
    if beta is None or beta == 1:
        mul_b = self
    else:
        mul_b = op.Mul(self, op.CastLike(beta, self))
```

**Pull request merged**